include/node.h

```diff
@@ -42,34 +42,64 @@ class Node {
 
   //! Assign hydraulic head at the node
   //! \param[in] head hydraulic head at the node
-  void head(double head) {
+  void elevation(double head) {
+    elevation_ = head;
     head_ = head;
     ishead_ = true;
   }
+  void head(double iter_head) { head_ = iter_head; }
 
   //! Return hydraulic head
   //! \retval head_ hydraulic head at the node
+  double elevation() const { return elevation_; }
   double head() const { return head_; }
 
   //! Return head assignment status
   //! \retval ishead_ head assignment status at the node
   bool ishead() const { return ishead_; }
+  //! Return if node is a reservoir
+  //! \retval isres_ reservoir status
+  bool isres() const { return isres_; }
 
   //! Assign discharge at the node
   //! \param[in] discharge discharge at the node
-  void discharge(double discharge) {
-    discharge_ = discharge;
+  void demand(double discharge) {
+    demand_ = discharge;
+      iter_demand_ = discharge;
     isdischarge_ = true;
+    // if demand is negative, it is a reservoir or tank
+    if (discharge < 0) {
+      isres_ = true;
+    }
   }
+  double iter_demand() { return iter_demand_; }
 
+  void iter_demand(double iter_discharge) { iter_demand_ = iter_discharge; }
   //! Return discharge
   //! \retval discharge_ discharge at the node
-  double discharge() const { return discharge_; }
+  double demand() const { return demand_; }
 
   //! Return discharge assignment status
   //! \retval isdischarge_ discharge assignment status at the node
   bool isdischarge() const { return isdischarge_; }
 
+  //! min pressure
+  void min_pressure(double min_pressure) { minimum_pressure_ = min_pressure; }
```

Comment:
Do we need all these getters and setters?

Reply:
I thought maybe people want to make different nodes to have difference min/norm pressures

Reply:
I'm happy with setting pressures, but having all variables with getters/setters means it breaks the philosophy of having classes

Reply:
Could we instead have a JSON or something format to read all geometrical and pressure properties of a pipe?

Reply:
if not maybe remove all unused getters